README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# Shiny Application for Spatial Sampling

<!-- badges: start -->

[![Project Status: Active – The project has reached a stable, usable
state and is being actively
developed.](https://www.repostatus.org/badges/latest/active.svg)](https://www.repostatus.org/#active)
![version](https://img.shields.io/badge/version-v0.3.0-blue)
[![license](https://img.shields.io/badge/license-MIT-blue)](https://github.com/ecohealthalliance/spatialsampling/blob/master/LICENSE)
<!-- badges: end -->

Area sampling approaches such as **centric systematic area sampling** or
**CSAS** have potential applications in studies that sample from mixed
human and animal subjects, and from the environment. Standard sampling
approaches that select subjects proportional to population size (PPS)
are impractical for use in these studies because sampling units selected
for one subject will not necessarily be representative of the other
subjects. This is particularly true in contexts where animal population
size distribution is inversely related to human population size.

With **CSAS**, sampling of any subject is performed systematically over
geographic area and resulting sample is said to be spatially
representative. This type of sample is said to approximate a simple
random sample (SRS)<sup>[1]</sup>. Thus, a mixed human and animal
subject study can use the same spatial sampling frame for all subjects.
Additional advantages of a **CSAS** approach is that the resulting
sample is implicitly spatially stratified which contributes to increased
sampling variance<sup>[2][3]</sup>.

## Steps in CSAS

### Step 1: Find a map

The first step is to find a map of the study area. Try to find a map
with as much detail and features (e.g., towns, villages, landmarks, etc)
as possible.

Below is a map of a study area somewhere in Tanzania.

<img src="figures/step1-1.png" style="display: block; margin: auto;" />

### Step 2: Draw a grid

The size of each square should be small enough for it to be reasonable
to assume homogeneity within the square. The size of the grid will also
be dictated by a target number of sampling units that you are aiming
for.

Below is an example of a grid overlaid onto the previous map to create
200 square grids

<img src="figures/step2-1.png" style="display: block; margin: auto;" />

### Step 3: Select the areas to sample

The sampling points, which are the centroids of the sampling grid,
indicates where to sample. For human and animal populations, this can be
the village or settlement or farm at or near the centroid location. For
environmental sampling, this would be specimens drawn at or near the
centroid locations.

<br>

## Data analysis considerations

Data collected from this type of sampling frame can the be made
population representative by applying a population weighted analysis
during indicator/outcome estimation. This can be done parametrically
using Taylor linearised deviation techniques (which can be implemented
in [R](https://cran.r-project.org) using Thomas Lumley’s [survey
package](https://cran.r-project.org/web/packages/survey/survey.pdf)<sup>[4]</sup>)
or using a non-parametric weighted bootstrap approach such as the one
described [here](https://github.com/rapidsurvys/bbw)<sup>[5]</sup>.

<br>

## About the `spatialsampling` Shiny application

This [Shiny](https://shiny.rstudio.com) application assists users in the
process of applying **CSAS** to a specified study area. This application
utilises the [R](https://cran.r-project.org) package
[`spatialsampler`](https://github.com/spatialworks/spatialsampler)<sup>[6]</sup>
which provides functions for performing **CSAS**.

The application can be accessed either via [EcoHealth
Alliance](https://www.ecohealthalliance.org)’s
[Shiny](https://shiny.rstudio.com) server, or locally by cloning this
[GitHub
repository](https://github.com/ecohealthalliance/spatialsampling).

### Accessing the Shiny application via the EHA server

To use the [Shiny](https://shiny.rstudio.com) application via [EcoHealth
Alliance](https://www.ecohealthalliance.org)’s
[Shiny](https://shiny.rstudio.com) server, go to
<https://connect.eha.io/spatialsampling>.

This is the most straightforward and easiest way of accessing the
[Shiny](https://shiny.rstudio.com) application.

### Accessing the Shiny application locally

To use the [Shiny](https://shiny.rstudio.com) application locally, you
will need to clone or download this [GitHub
repository](https://github.com/ecohealthalliance/spatialsampling) into
your local computer.

The figure below shows how to download this [GitHub
repository](https://github.com/ecohealthalliance/spatialsampling):

<br/>

<img src="figures/downloadGitHubRepo.png" width="2880" style="display: block; margin: auto;" />

<br/>

The repository will then be downloaded as a `.zip` file into your
default downloads folder. Unzip the contents of the `.zip` file into
your preferred local directory.

To clone this [GitHub
repository](https://github.com/ecohealthalliance/spatialsampling) via
`https`, type the following command in your terminal:

``` shell
git clone https://github.com/ecohealthalliance/spatialsampling.git
```

or via `ssh`:

``` shell
git clone git@github.com:ecohealthalliance/spatialsampling.git
```

Once you have downloaded or cloned the repository locally, open the
folder and then double-click the `spatialsampling.Rproj` file to open
the project in [RStudio](https://rstudio.com). Once on
[RStudio](https://rstudio.com), run the application using the following
command in the [R](https://cran.r-project.org) console:

``` r
if (!require(shiny)) install.packages("shiny")

shiny::runApp(launch.browser = TRUE)
```

Please note that there are [R](https://cran.r-project.org) packages
required for running the [Shiny](https://shiny.rstudio.com) application
that you may have to install if they are not already installed on your
local machine. This will most likley be the main issue that will arise
when running this application locally. Please install missing packages
as indicated in the resulting error messages (if any). Note also that
installation of [R](https://cran.r-project.org) spatial packages can be
tricky and a bit complicated.

This approach of accessing the [Shiny](https://shiny.rstudio.com)
application is more suitable for those who are more familiar with
running [Shiny](https://shiny.rstudio.com) applications locally, those
who would like to see the source code of the application and possibly
contribute to it via
[GitHub](https://github.com/ecohealthalliance/spatialsampling), and/or
those who are familiar and comfortable with installing
[R](https://cran.r-project.org) spatial packages. We recommend using
this second approach if you fall within the categories described above
or if you are up for a challenge!

### Using the Shiny application

You will see the following on your web browser when you access the
[Shiny](https://shiny.rstudio.com) application either remotely on
[EHA](https://www.ecohealthalliance.org)’s
[Shiny](https://shiny.rstudio.com) server or locally through your copy
of the application.

<br>

<img src="figures/spatialSampling.png" width="1000" style="display: block; margin: auto;" />

<br>

The basic interface of the [Shiny](https://shiny.rstudio.com)
application is that of a map with an overlay menu on the upper right
hand corner where the user inputs a few required parameters to be used
for spatial sampling. When the menu is not in use, it turns transparent.
Hover your cursor over the menu and it will turn opaque allowing for the
menu buttons to be readable.

The following points describe the different input parameters for the
[Shiny](https://shiny.rstudio.com) application. These can also be found
as a quick start-up guide modal on the
[Shiny](https://shiny.rstudio.com) application itself which can be
accessed by clicking on the information link on the overlay menu.

#### Select study country

Select country where study is to be done.

Please note that once you have selected a country, the application will
download and read in the map boundary files for the specified country.

#### Upload study area map

Upload study area map in either shapefiles (SHP) or geopackage (GPKG)
format. If uploading shapefiles, select a zip file of a folder
containing the multiple files required.

To trial the application, the following sample study area maps for
specific countries can be downloaded and used for this step:

-   Bangladesh -
    [survey\_area\_bgd.gpkg](https://github.com/ecohealthalliance/spatialsampling/blob/main/maps/survey_area_bgd.gpkg?raw=true)

-   Jordan -
    [survey\_area\_jor.gpkg](https://github.com/ecohealthalliance/spatialsampling/blob/main/maps/survey_area_jor.gpkg?raw=true)

-   Liberia -
    [survey\_area\_lbr.gpkg](https://github.com/ecohealthalliance/spatialsampling/blob/main/maps/survey_area_lbr.gpkg?raw=true)

-   Philippines -
    [survey\_area\_phl.gpkg](https://github.com/ecohealthalliance/spatialsampling/blob/main/maps/survey_area_phl.gpkg?raw=true)

-   South Africa -
    [survey\_area\_zaf.gpkg](https://github.com/ecohealthalliance/spatialsampling/blob/main/maps/survey_area_zaf.gpkg?raw=true)

-   Tanzania -
    [survey\_area\_tza.gpkg](https://github.com/ecohealthalliance/spatialsampling/blob/main/maps/survey_area_tza.gpkg?raw=true)

Once a map file has been uploaded, this will be read and processed. Part
of this processing includes the subsetting of the country administrative
boundaries to those that are local to the study area map provided, and
the extraction of human and cattle population rasters appropriate for
the study area. When this process has completed, the study area and
administrative boundaries along with the human population raster will be
shown on the map while the cattle population raster is hidden but can be
activated via the layers menu on the bottom left side of the map.

#### Number of sampling units

Input the number of sampling units required or desired. For a human
population study, the recommended minimum number of sampling units is at
least 30 (default). This is the minimum number of sampling clusters
recommended to achieve sampling variance that will provide
proportion/prevalence estimates with ±3-5% precision. This should be
adjusted accordingly based on study’s target sample size and target
cluster size.

#### Set a sampling area buffer

Sometimes, it is necessary to add a buffer around the study area to be
able to spread the spatial sample evenly up to the edges. This buffer
can be specified here. By default, this is set to 0 for no buffer. The
buffer should be specified in kilometres (kms). This parameter is
optional but as a rule-of-thumb, it is recommended to add a buffer as
the study area increases and/or if your number of sampling units is
closer to 30.

#### Select centric or eccentric sample

By default, the application will perform a *centric sample* selection
which is signified by the tick on the checkbox labeled “Centric sample”.
If desired, an *eccentric sample* can be selected (sampling point is
randomly selected at any part of each of the sampling grids rather than
just the centre). Untick this checkbox if an *eccentric sample* is
desired.

#### Sample

Once the required number of sampling units has been specified, a spatial
sample can now be taken.

Click the sampling button and a sampling grid and sampling points based
on the specifications will be added to the map.

#### List

Once the sampling grid and the sampling points have been generated on
the map, a list of the sampling points and their identifying features,
geocoordinates, and human and cattle population weights is generated and
can be saved and downloaded by clicking this button.

#### Refresh

If you want to redo the spatial sampling for some reason or if you want
to do a spatial sample for a different country or a different study area
within the same country, click on the refresh button and then repeat the
steps above.

<br>

## Contributors

This [Shiny](https://shiny.rstudio.com) application was made possible by
contributions from:

-   [Ernest Guevarra](mailto:guevarra@ecohealthalliance.org)

-   [Melinda Rostal](mailto:rostal@ecohealthalliance.org)

-   [Rebecca Bodenham](mailto:bodenham@ecohealthalliance.org)

<br>

## License

This [Shiny](https://shiny.rstudio.com) application is made available
under a [MIT
license](https://github.com/ecohealthalliance/spatialsampling/blob/master/LICENSE).

<br>

## Issues and feedback

Feedback, bug reports and feature requests are welcome; file issues or
seek support
[here](https://github.com/ecohealthalliance/spatialsampling/issues).

<br>

## References

[1] Milne, A. (1959). The Centric Systematic Area-Sample Treated as a
Random Sample. Biometrics, 15(2), 270-297. <doi:10.2307/2527674>

[2] Aaron GJ, Strutt N, Boateng NA, Guevarra E, Siling K, et al. (2016)
Assessing Program Coverage of Two Approaches to Distributing a
Complementary Feeding Supplement to Infants and Young Children in Ghana.
PLOS ONE 11(10): e0162462.
<https://doi.org/10.1371/journal.pone.0162462>

[3] Aaron, G. J. et al. (2016) ‘Household coverage of fortified staple
food commodities in Rajasthan, India’, PLoS ONE, 11(10).
<https://doi.org/10.1371/journal.pone.0163176>

[4] Lumley T. Analysis of complex survey samples. Journal of Statistical
Software. 2004;9: 1–19. Available:
<http://www.jstatsoft.org/v09/a08/paper>

[5] Mark Myatt (2018). bbw: Blocked Weighted Bootstrap. R package
version 0.1.3. <https://CRAN.R-project.org/package=bbw>

[6] Mark Myatt, Farah Ibrahim and Ernest Guevarra (2018).
spatialsampler: An Implementation of the Centric Systematic Area
Sampling (CSAS) and Simple Spatial Sampling Method (S3M) sampling
approaches in R. R package version 0.1.0.
<https://github.com/spatialworks/spatialsampler>


NEWS.md
# `spatialsampling` v0.3.0

Third release of `spatialsampling`. In this release:

* Added function to perform *eccentric sample* selection within the sampling grids.

* Updated README to reflect usage of new function.


# `spatialsampling` v0.2.0

Second release of `spatialsampling`. In this release:

* Published the application to EHA's new RSConnect server via GitHub using a `manifest.json`.

* Updated documentation to reflect new application URL.

* Updated some package dependencies given changes in sources/repositories of some packages.

* Added a NEWS.md to track application versions.


# `spatialsampling` v0.1.0

First release of `spatialsampling`. In this release:

* Countries included in application are Bangladesh, Jordan, Liberia, Philippines, South Africa, and Tanzania.

* Human population data is drawn from population maps provided by [WorldPop](https://www.worldpop.org).

* Cattle population data is drawn from the [Global Livestock of the World 3 (GLW3)](https://dataverse.harvard.edu/dataverse/glw_3) dataset.

* Outputs a spreadsheet (Microsoft Excel) of list of sampling locations with features, geographic coordinates and human and cattle population weights.

* Published the application on EHA's Shinyapp server at https://aegypti.ecohealthalliance.org/shiny/guevarra/spatialsampling




ui.R
################################################################################
#
# UI
#
################################################################################

## Set navbar
navbarPage(title = "Spatial Sampling", id = "chosenTab",
  ## Create tab panel
  tabPanel(title = "",
    value = 1,
    icon = icon(name = "home", class = "fa-lg", lib = "font-awesome"),
    ## Header HTML
    div(class = "outer",
      ## Header HTML
      tags$head(
        tags$link(rel = "stylesheet", type = "text/css", href = "styles.css")
      ),
      ## Output map
      leafletOutput("map", width = "100%", height = "100%"),
      ## Absolute panel - input
      absolutePanel(id = "controls",
        class = "panel panel-default",
        fixed = TRUE, draggable = FALSE,
        top = 65, left = "auto", right = 10, bottom = "auto",
        width = 330, height = "auto",
        ## shinyjs
        useShinyjs(),
        ## UI header
        div(style="display: inline-block;vertical-align: middle;",
          h4("Sampling inputs and parameters")
        ),
        ## Action link for modal
        div(style="display: inline-block;vertical-align: middle;",
          actionLink(inputId = "inputs_info",
            label = "",
            icon = icon(name = "info-sign", lib = "glyphicon")
          )
        ),
        ##
        selectInput(inputId = "country",
          label = "Study country",
          choices = c("Select study country" = " ",
                      "Bangladesh", "Jordan", "Liberia",
                      "Philippines", "South Africa", "Tanzania"),
          selected = " "
        ),
        ## Upload study area map
        fileInput(inputId = "survey_map",
          label = "Upload file of study area map",
          accept = c(".gpkg", ".zip")
        ),
        #uiOutput("input_map_file"),
        ## Horizontal line break
        hr(),
        ## Input sampling parameters
        uiOutput("sample_parameters1"),
        uiOutput("sample_parameters2"),
        uiOutput("sample_parameters4"),
        div(style="display:inline-block",
          uiOutput("sample_parameters3")
        ),
        ## Report output
        #div(style="display:inline-block",
        #  uiOutput("report_output")
        #),
        ## List output
        div(style="display:inline-block",
          uiOutput("list_output")
        ),
        ## Reset button
        div(style="display:inline-block",
          uiOutput("reset_sample")
        )
      )
    )
  ),
  ## Settings page
  tabPanel(title = "Settings",
    value = 2,
    icon = icon(name = "cog", class = "fa-lg", lib = "font-awesome"),
    tabsetPanel(
      tabPanel(title = "Datasets",
        fluidPage(
          useShinyjs(),
          fluidRow(
            ## add empty row for spacing
            column(width = 12, br())
          ),
          fluidRow(
            column(width = 3,
              wellPanel(id = "population_dataset_settings",
                h4("Population"),
                tags$p("This application uses pre-loaded population datasets
                       for humans and cattles retrieved from the following
                       sources:"),
                radioButtons(inputId = "pop_data",
                  label = "",
                  choices = c("WorldPop" = "worldpop",
                              "GLW3" = "glw3"),
                  inline = TRUE
                ),
                hr(),
                h5("Upload population datasets"),
                tags$p("Alternative sources of population datasets for humans
                       and/or cattle can be specified by uploading these
                       sources here."),
                ## Select alternative input
                selectInput(inputId = "dataset_type",
                  label = "Alternative population dataset for",
                  choices = c("humans", "cattles"),
                  selected = "humans"
                ),
                selectInput(inputId = "input_type_alt",
                  label = "Select alternative dataset file type",
                  choices = c("shapefile" = "shp",
                              "geopackage" = "gpkg"),
                  selected = c("shp")
                ),
                uiOutput("dataset_input")
              )
            ),
            column(width = 3,
              wellPanel(id = "boundary_dataset_settings",
                h4("Country boundaries"),
                HTML("<p>This application uses boundary files available from
                     <a href='https://gadm.org'>https://gadm.org</a>.</p>"),
                br(), br(), br(),
                hr(),
                h5("Upload map boundaries dataset"),
                tags$p("Alternative sources of boundaries datasets can be
                        specified by uploading here."),
                selectInput(inputId = "input_type_boundaries",
                  label = "Select alternative boundaries file type",
                  choices = c("shapefile" = "shp",
                              "geopackage" = "gpkg"),
                  selected = c("shp")
                ),
                uiOutput("boundaries_input"),
                br(), br(), br(), br(), br()
              )
            )
          )
        )
      ),
      tabPanel(title = "Maps",
        fluidPage(
          useShinyjs(),
          fluidRow(
            ## add empty row for horizontal spacing
            column(width = 12, br())
          ),
          fluidRow(
            column(width = 3,
              wellPanel(id = "map_layers_settings_panel",
                h4("Base map layers"),
                ## Select base layer
                selectInput(inputId = "base_layer",
                  label = "Select base layer",
                  choices = c("Satellite Street" = "satellite",
                              "Le Shine" = "leShine",
                              "Decimal" = "decimal",
                              "Standard" = "standard",
                              "Northstar" = "northstar",
                              "Terminal" = "teminal",
                              "Moonlight" = "moonlight"),
                  selected = "satellite"
                ),
                hr(),
                ## Show sample of selected base layer
                leafletOutput("sample_base_map", height = 300)
              )
            ),
            column(width = 3,
              wellPanel(id = "boundaries_settings_panel",
                h4("Area boundaries"),
                ## Select colour for study area boundaries
                colourpicker::colourInput(inputId = "survey_area_colour",
                  label = "Colour of survey area boundaries",
                  value = "green",
                  palette = "limited",
                  allowTransparent = TRUE,
                  returnName = TRUE
                ),
                ## Select weight of study area boundaries
                numericInput(inputId = "survey_area_weight",
                  label = "Weigth of survey area boundaries",
                  value = 5,
                  min = 5, max = 20,
                  step = 1
                ),
                hr(),
                ## Select colour for country area boundaries
                colourpicker::colourInput(inputId = "country_boundaries_colour",
                  label = "Colour of country boundaries",
                  value = "yellow",
                  palette = "limited",
                  allowTransparent = TRUE,
                  returnName = TRUE
                ),
                ## Select weight of country area boundaries
                numericInput(inputId = "country_boundaries_weight",
                  label = "Weight of country area boundaries",
                  value = 2,
                  min = 5, max = 20,
                  step = 1
                ),
                hr(),
                ## Reset selection
                div(style="display:inline-block; vertical-align:middle;",
                  actionButton(inputId = "reset_boundary_settings",
                    label = "Reset",
                    class = "btn-primary",
                    icon = icon(name = "refresh",
                                class = "fa-med",
                                lib = "font-awesome"))
                ),
                ## Save centroid settings
                div(style="display:inline-block; vertical-align:middle;",
                  actionButton(inputId = "save_boundary_settings",
                    label = "Save",
                    class = "btn-success",
                    icon = icon(name = "check",
                                class = "fa-med",
                                lib = "font-awesome"))
                )
              )
            ),
            column(width = 3,
              wellPanel(id = "sample_points_settings_panel",
                h4("Sampling points"),
                ## Select colour for sampling points
                colourpicker::colourInput(inputId = "centroid_colour",
                  label = "Colour of sampling points",
                  value = "red",
                  palette = "limited",
                  allowTransparent = TRUE,
                  returnName = TRUE
                ),
                ## Select size for sampling points
                numericInput(inputId = "centroid_size",
                  label = "Size of sampling points",
                  value = 2,
                  min = 1, max = 10, step = 1
                ),
                ## Select weight for sampling points
                numericInput(inputId = "centroid_weight",
                  label = "Weight of sampling points",
                  value = 5,
                  min = 5, max = 10, step = 1
                ),
                br(), br(), br(), br(), br(),
                hr(),
                ## Reset selection
                div(style="display:inline-block; vertical-align:middle;",
                  actionButton(inputId = "reset_centroid_settings",
                    label = "Reset",
                    class = "btn-primary",
                    icon = icon(name = "refresh",
                                class = "fa-med",
                                lib = "font-awesome")
                  )
                ),
                ## Save centroid settings
                div(style="display:inline-block; vertical-align:middle;",
                  actionButton(inputId = "save_centroid_settings",
                    label = "Save",
                    class = "btn-success",
                    icon = icon(name = "check",
                                class = "fa-med",
                                lib = "font-awesome"))
                )
              )
            ),
            column(width = 3,
              wellPanel(id = "sample_grid_settings_panel",
                h4("Sampling grid"),
                ## Select colour for sample grid
                colourpicker::colourInput(inputId = "grid_colour",
                  label = "Colour of sampling grid lines",
                  value = "blue",
                  palette = "limited",
                  allowTransparent = TRUE,
                  returnName = TRUE
                ),
                ## Select weight for sampling grid
                numericInput(inputId = "grid_weight",
                  label = "Weight of sampling grid lines",
                  value = 5,
                  min = 5, max = 10, step = 1
                ),
                br(), br(), br(), br(), br(), br(), br(), br(), br(),
                hr(),
                ## Reset selection
                div(style="display:inline-block; vertical-align:middle;",
                  actionButton(inputId = "reset_grid_settings",
                    label = "Reset",
                    class = "btn-primary",
                    icon = icon(name = "refresh",
                                class = "fa-med",
                                lib = "font-awesome")
                  )
                ),
                ## Save centroid settings
                div(style="display:inline-block; vertical-align:middle;",
                  actionButton(inputId = "save_grid_settings",
                    label = "Save",
                    class = "btn-success",
                    icon = icon(name = "check",
                                class = "fa-med",
                                lib = "font-awesome"))
                )
              )
            )
          )
        )
      )
    )
  ),
  ## About page
  tabPanel(title = "About",
    value = 3,
    icon = icon(name = "info-circle", class = "fa-lg", lib = "font-awesome"),
    sidebarPanel(
      width = 4,
      HTML(
        "
        <h4>The <code>spatialsampling</code> Shiny application has been developed
        specifically for EcoHealth Alliance's Crimean-Congo Haemorrhagic Fever
        (CCHF) project in Tanzania.</h4>
        <br>
        <h4>However, this application has potential uses for any other project
        as an alternative to traditional population proportional sampling (PPS)
        sampling frames.</h4>
        <br>
        <h4>To learn more about spatial sampling and for using it in your study
        or project, please contact <a href='mailto:guevarra@ecohealthalliance.org'>
        Ernest Guevarra</a>.</h4>
        "
      )
    ),
    mainPanel(width = 8,
      HTML(
        "
        <h2>Spatial sampling</h2>
        <p>Area sampling approaches such as <strong>centric systematic area
        sampling</strong> or <strong>CSAS</strong> have potential applications in
        studies that sample from mixed human and animal subjects, and from the
        environment. Standard sampling approaches that select subjects
        <em>proportional to population size (PPS)</em> are impractical for use in
        these studies because sampling units selected for one subject will not
        necessarily be representative of the other subjects. This is particularly
        true in contexts where animal population size distribution is inversely
        related to human population size.</p>
        <br>
        <p>With <strong>CSAS</strong>, sampling of any subject is performed
        systematically over geographic area and resulting sample is said to be
        spatially representative. This type of sample is approximates a
        <strong>simple random sample (SRS)</strong><sup>1</sup>. Thus, a mixed
        human and animal subject study can use the same spatial sampling frame
        for all subjects. Additional advantages of a <strong>CSAS</strong>
        approach is that the resulting sample is implicitly spatially stratified
        which contributes to increased sampling variance<sup>2,3</sup>.</p>
        <br>
        <h3>Steps in CSAS</h3>
        <br>
        <h4>Step 1: Find a map</h4>
        <p>The first step is to find a map of the study area. Try to find a map
        with as much detail and features (e.g., towns, villages, landmarks, etc)
        as possible.</p>
        <br>
        <h4>Step 2: Draw a grid</h4>
        <p>The size of each square should be small enough for it to be reasonable
        to assume homogeneity within the square. The size of the grid will also
        be dictated by a target number of sampling units that you are aiming
        for.</p>
        <br>
        <h4>Step 3: Select the areas to sample</h4>
        <p>The sampling points which are the centroids of the sampling grid points
        us where to sample. For human and animal populations, this can be the
        village or settlement at or near the centroid location. For environmental
        sampling, this would be specimens drawn at or near the centroid
        locations.</p>
        <br>
        <h3>Data analysis considerations</h3>
        <p>Data collected from this type of sampling frame can the be made
        population representative by applying a population weighted analysis
        during indicator/outcome estimation. This can be done parametrically
        using Taylor linearised deviation techniques (which can be implemented in
        <a href='https://cran.r-project.org' target='_blank'>R</a> using Thomas
        Lumley's <a href='https://cran.r-project.org/web/packages/survey/survey.pdf'
        target='_blank'>survey package</a><sup>4</sup> or using a non-parametric
        weighted bootstrap approach such as the one described
        <a href='https://github.com/rapidsurvys/bbw' target='_blank'>here</a>
        <sup>5</sup>.</p>
        <br>
        <h3>The Shiny application</h3>
        <p>This <a href='http://shiny.rstudio.com' target='_blank'>Shiny</a>
        application assists users in the process of applying <strong>CSAS</strong>
        to a specified study area. This application utilises the
        <a href='https://cran.r-project.org' target='_blank'>R</a> package
        <a href='https://github.com/spatialworks/spatialsampler'
        target='_blank'>spatialsampler</a><sup>6</sup> which provides functions
        for performing <strong>CSAS</strong>.</p>
        <br>
        <h3>Contributors</h3>
        <p>This <a href='https://shiny.rstudio.com' target='_blank'>Shiny</a>
        application was made possible by contributions from:
        <ul>
          <li><a href='mailto:guevarra@ecohealthalliance.org'>Ernest Guevarra</a>
          <li><a href='mailto:rostal@ecohealthalliance.org'>Melinda Rostal</a>
          <li><a href='mailto:bodenham@ecohealthalliance.org'>Rebecca Bodenham</a>
        </ul>
        <br>
        <h3>License</h3>
        <p>This <a href='https://shiny.rstudio.com target='_blank'>Shiny</a>
        application is available under a <a href='https://github.com/ecohealthalliance/spatialsampling/blob/master/LICENSE'
        target='_blank'>MIT license</a>.</p>
        <br>
        <h3>Issues and feedback</h3>
        <p>Feedback, bug reports and feature requests are welcome; file issues
        or seek support <a href='https://github.com/ecohealthalliance/spatialsampling/issues'
        target='_blank'>here</a></p>
        <br>
        <h3>References</h3>
        <p><sup>1</sup> Milne, A. (1959). The Centric Systematic Area-Sample
        Treated as a Random Sample. Biometrics, 15(2), 270-297.
        <a href='https://doi.org/10.2307/2527674'
        target='_blank'>https://doi.org/10.2307/2527674</a></p>
        <p><sup>2</sup> Aaron GJ, Strutt N, Boateng NA, Guevarra E, Siling K,
        et al. (2016) Assessing Program Coverage of Two Approaches to
        Distributing a Complementary Feeding Supplement to Infants and Young
        Children in Ghana. PLOS ONE 11(10): e0162462.
        <a href='https://doi.org/10.1371/journal.pone.0162462' target='_blank'>
        https://doi.org/10.1371/journal.pone.0162462</a></p>
        <p><sup>3</sup> Aaron, G. J. et al. (2016) ‘Household coverage of fortified
        staple food commodities in Rajasthan, India’, PLoS ONE, 11(10).
        <a href='https://doi.org/10.1371/journal.pone.0163176' target='_blank'>
        https://doi.org/10.1371/journal.pone.0163176</a></p>
        <p><sup>4</sup> Lumley T. Analysis of complex survey samples. Journal of
        Statistical Software. 2004;9: 1–19. Available:
        <a href='http://www.jstatsoft.org/v09/a08/paper' target='_blank'>
        http://www.jstatsoft.org/v09/a08/paper</a></p>
        <p><sup>5</sup> Mark Myatt (2018). bbw: Blocked Weighted Bootstrap. R
        package version 0.1.3. <a href='https://CRAN.R-project.org/package=bbw'
        target='_blank'>https://CRAN.R-project.org/package=bbw</a></p>
        <p><sup>6</sup> Mark Myatt, Farah Ibrahim and Ernest Guevarra (2021).
        spatialsampler: An Implementation of the Centric Systematic Area Sampling
        (CSAS) and Simple Spatial Sampling Method (S3M) sampling approaches in R.
        R package version 0.1.0. <a href='https://github.com/spatialworks/spatialsampler'
        target='_blank'>https://github.com/spatialworks/spatialsampler</a></p>
        "
      )
    )
  )
)


